riscv/runtime/src/coprocessors.rs

```diff
@@ -0,0 +1,10 @@
+// This is a dummy implementation of Poseidon hash,
+// which will be replaced with a call to the poseidon
+// coporocessor during compilation.
+// The function itself will be removed by the compiler
+// during the reachability analysis.
+#[no_mangle]
```

Comment:
I'm wondering if declaring this as a real FFI function would be better:
```
extern "C" {
    pub fn poseidon_hash(a: u32, b: u32) -> u32;
}
```

Reply:
just pushed a commit that does that